pinot-spi/src/main/java/org/apache/pinot/spi/utils/GzipUtils.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+
+
+/**
+ * Utils class for Gzip related utilities.
+ */
+public class GzipUtils {
+  // Disable instantiation
+  private GzipUtils() {
+  }
+
+  /**
+   * Utility method to identify if a file is gzipped.
+   *
+   * @param file True if file exists and is gzipped.
+   * @return False, otherwise
+   */
+  public static boolean isGZipped(File file) {
+    int magic;
+    try {
+      InputStream inputStream = new FileInputStream(file);
+      magic = inputStream.read() & 0xff | ((inputStream.read() << 8) & 0xff00);
+      inputStream.close();
```

Comment:
Do try with resource to ensure the stream is closed
```suggestion
    try (InputStream inputStream = new FileInputStream(file)) {
      magic = inputStream.read() & 0xff | ((inputStream.read() << 8) & 0xff00);
```